### ISSUE TITLE
bug: returned kpoint in get_eigenmode_coefficients

### ISSUE THREAD

**Issue by oskooi** [stevengj/meep#438]
There is a bug in `get_eigenmode_coefficients` having to do with the kpoints which are returned. This is demonstrated in the script below which sets the Bloch periodic wavevector (`k_point`) to an arbitrary value of `(0.1,0.2,0)` for an empty 2d computational cell. At the end of the simulation, the kpoints obtained from calling `get_eigenmode_coefficients` are printed to standard output.

```py
import meep as mp

resolution = 20
dpml = 1
sx = 10
sy = 5

cell_size = mp.Vector3(sx,sy,0)
pml_layers = [mp.PML(thickness=dpml,direction=mp.X)]

fcen = 1.0
sources = [mp.Source(mp.GaussianSource(fcen,fwidth=0.2),
                     component=mp.Ez,
                     center=mp.Vector3(-0.5*sx+dpml+0.5),
                     size=mp.Vector3(0,sy))]

sim = mp.Simulation(resolution=resolution,
                    cell_size=cell_size,
                    boundary_layers=pml_layers,
                    k_point=mp.Vector3(0.1,0.2,0),
                    sources=sources)

mode_mon_pt = mp.Vector3(0.5*sx-dpml-0.5)
mode_mon = sim.add_mode_monitor(fcen, 0, 1, mp.FluxRegion(center=mode_mon_pt, size=mp.Vector3(0,sy,0)))

sim.run(until_after_sources=mp.stop_when_fields_decayed(50, mp.Ez, mode_mon_pt, 1e-8))

for n in range(5):
  coeffs, vgrps, kpoints = sim.get_eigenmode_coefficients(mode_mon, [n+1], eig_parity=mp.ODD_Z)
  print("grating{}:, {:.6f}, {:.6f}".format(n,kpoints[0].x,kpoints[0].y))
```
The output from running this script is shown below. MPB is computing modes with a y-component of the wavevector of `0.2`. However, the kpoints returned from `get_eigenmode_coefficients` have a y-component of the wavevector of `0.04`.

**Output**

```
Using MPI version 3.0, 1 processes
-----------
Initializing structure...
Working in 2D dimensions.
Computational cell is 10 x 5 x 0 with resolution 20
time for set_epsilon = 0.0504379 s
-----------
Meep: using complex fields.
field decay(t = 50.025000000000006): 0.2558033294701495 / 0.2558033294701495 = 1.0
field decay(t = 100.05000000000001): 6.124990782353389e-07 / 0.2558033294701495 = 2.3944140191764522e-06
field decay(t = 150.07500000000002): 1.0851944275031765e-13 / 0.2558033294701495 = 4.2422998549352786e-13
run 0 finished at t = 150.07500000000002 (6003 timesteps)
MPB solved for omega_1(1,0.2,0) = 1.0008 after 21 iters
MPB solved for omega_1(0.9992,0.2,0) = 1 after 1 iters
grating0:, 0.999200, 0.040000
MPB solved for omega_2(1,0.2,0) = 1.01272 after 23 iters
MPB solved for omega_2(0.987119,0.2,0) = 1 after 1 iters
MPB solved for omega_2(0.987117,0.2,0) = 1 after 1 iters
grating1:, 0.987117, 0.040000
MPB solved for omega_3(1,0.2,0) = 1.0284 after 21 iters
MPB solved for omega_3(0.970797,0.2,0) = 1.00002 after 1 iters
MPB solved for omega_3(0.970773,0.2,0) = 1 after 1 iters
grating2:, 0.970773, 0.040000
MPB solved for omega_4(1,0.2,0) = 1.06283 after 17 iters
MPB solved for omega_4(0.933226,0.2,0) = 1.00026 after 1 iters
MPB solved for omega_4(0.932952,0.2,0) = 1 after 1 iters
grating3:, 0.932952, 0.040000
MPB solved for omega_5(1,0.2,0) = 1.09252 after 15 iters
MPB solved for omega_5(0.89892,0.2,0) = 1.00083 after 1 iters
MPB solved for omega_5(0.897998,0.2,0) = 1 after 1 iters
grating4:, 0.897998, 0.040000

Elapsed run time = 2.5906 s
```

**Comment by stevengj** [comment]
I think you just need to change mpb.cpp lines 281–285 to
```
  for (int i = 0; i < 3; ++i) {
    n[i] = int(resolution * s[i] + 0.5); if (n[i] == 0) n[i] = 1;
    R[i][i] = s[i] = s[i] == 0 ? 1 : s[i];
    G[i][i] = 1 / R[i][i]; // recip. latt. vectors / 2 pi
    k[i] *= R[i][i]; // convert k to reciprocal basis
  }
```

**Comment by stevengj** [comment]
My output now is:
```
run 0 finished at t = 150.075 (6003 timesteps)
MPB solved for omega_1(1,0.2,0) = 1 after 17 iters
grating0:, 1.000000, 0.200000
MPB solved for omega_2(1,0.2,0) = 1.0198 after 20 iters
MPB solved for omega_2(0.979804,0.2,0) = 1.00001 after 1 iters
MPB solved for omega_2(0.979796,0.2,0) = 1 after 1 iters
grating1:, 0.979796, 0.200000
MPB solved for omega_3(1,0.2,0) = 1.0198 after 16 iters
MPB solved for omega_3(0.979804,0.2,0) = 1.00001 after 1 iters
MPB solved for omega_3(0.979796,0.2,0) = 1 after 1 iters
grating2:, 0.979796, 0.200000
MPB solved for omega_4(1,0.2,0) = 1.07703 after 14 iters
MPB solved for omega_4(0.917033,0.2,0) = 1.00047 after 1 iters
MPB solved for omega_4(0.916515,0.2,0) = 1 after 1 iters
grating3:, 0.916515, 0.200000
MPB solved for omega_5(1,0.2,0) = 1.07703 after 14 iters
MPB solved for omega_5(0.917033,0.2,0) = 1.00047 after 1 iters
MPB solved for omega_5(0.916515,0.2,0) = 1 after 1 iters
grating4:, 0.916515, 0.200000
```
